service/history/events/notifier.go

```diff
@@ -111,9 +112,9 @@ func NewNotification(
 		LastFirstEventTxnID:    lastFirstEventTxnID,
 		NextEventID:            nextEventID,
 		PreviousStartedEventID: previousStartedEventID,
-		CurrentBranchToken:     currentBranchToken,
 		WorkflowState:          workflowState,
 		WorkflowStatus:         workflowStatus,
+		VersionHistories:       versionHistories,
```

Comment:
make a copy? or this will lead to data corruption

Reply:
i would suggest using last event ID & event version (last version history item)

Reply:
yes. I prefer last version history item. However, the PollMutableStateResponse contains the version histories.